multiqc/modules/samtools/rmdup.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Samtools rmdup """
+
+import logging
+import re
+from collections import OrderedDict, defaultdict
+from multiqc import config, plots
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class RmdupReportMixin():
+
+    def parse_samtools_rmdup(self):
+        """ Find Samtools rmdup logs and parse their data """
+
+        self.samtools_rmdup = dict()
+        for f in self.find_log_files(config.sp['samtools']['rmdup']):
+            parsed_data = parse_single_report(f['f'])
+            if len(parsed_data) > 0:
+                if f['s_name'] in self.samtools_rmdup:
+                    log.debug("Duplicate sample name found! Overwriting: {}".format(f['s_name']))
+                self.add_data_source(f, section='rmdup')
+                self.samtools_rmdup[f['s_name']] = parsed_data
+
+        if len(self.samtools_rmdup) > 0:
+
+            # Write parsed report data to a file (restructure first)
+            self.write_data_file(self.samtools_rmdup, 'multiqc_samtools_rmdup')
+
+        return len(self.samtools_idxstats)
+
+def parse_single_report(f):
```

Comment:
If you want to access the `self` variables in this object, you need to change this to:
```python
def parse_single_report(self, f):
```

(doesn't change how you call the function - it will still accept a single argument).